### DISCUSSION TITLE
Welcome to beaverhabits Discussions!

### DISCUSSION THREAD

**Discussion by daya0576** [daya0576/beaverhabits#17] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ocean-haiyang** [comment]
Loving your application so far! There is no satisfactory self hosted habit app until yours! Do you think you plan to stick to habits or consider adding to-dos? A self hosted ticktick would be awesome. It adds to-dos in one tab and habits in another. Also has a calendar and all three sync with caldav.

**Reply by daya0576** [reply]
Happy to hear that you do enjoy this app :) 

> Do you think you plan to stick to habits or consider adding to-dos?

Currently, the project focuses on minimalism. 

Could you share an example of the todo item, is it a one-time task? Maybe we can introduce todo tasks as a new type of habit.

> Also has a calendar and all three sync with caldav.

"Caldav" is new to me. Can you give an example of an expected workflow using it? Thanks!

**Reply by ocean-haiyang** [reply]
Maybe it would need to be a fork if this is not minimalist enough! I wish I could code better. 

https://ticktick.com/features This is the app but I don't like closed source. Your current direction for habits it awesome. If you were to add to-dos I would imagine, similar to this app, there would be two buttons on the bottom screen. One for habits then one for tasks. Tasks would keep the same list style as habits. It would be one-time tasks. Maybe also allow adding a date and time or simply just tasks for the day.

Caldav is a standard for syncing calendars but also allows tasks. https://en.wikipedia.org/wiki/CalDAV This could allow users to sync there tasks with their main caldav server. For instance I am using baikal for caldav and carddav.

![Screenshot from 2024-11-09 11-23-14](https://github.com/user-attachments/assets/68640ab1-f9dd-49d0-a091-6f4d8eb0e0b3)
 Something like my horrible rendering, lol

**Reply by ocean-haiyang** [reply]
I blurred my habits. But the Tasks button could lead to a similar page for one off tasks. You can create and or complete them.
